internal/driver/fetch_test.go

```diff
@@ -423,9 +423,15 @@ func TestHttpsInsecure(t *testing.T) {
 	if len(p.SampleType) == 0 {
 		t.Fatalf("fetchProfiles(%s) got empty profile: len(p.SampleType)==0", address)
 	}
-	if runtime.GOOS == "plan9" {
+	switch runtime.GOOS {
+	case "plan9":
 		// CPU profiling is not supported on Plan9; see golang.org/issues/22564.
 		return
+	case "darwin":
+		if runtime.GOARCH == "arm" || runtime.GOARCH == "arm64" {
+			// CPU profiling is not supported on OS; see golang.org/issues/22612.
```

Comment:
Is CPU profiling not supported on iOS in the same way as on plan9, or the quality of profiles is low and there are no symbolized function names in the profile? Looking at https://github.com/golang/go/blob/83a1a2ba63cd79ed65a97bead2c31ef0753f4cd2/src/runtime/pprof/pprof_test.go#L148, I don't see the CPU profile test being skipped in the Go tests?

Reply:
I think CPU profiling works on iOS, so my guess would be missing symbols, judging from the failure log:

````
Local symbolization failed for cppbench_server_main: stat cppbench_server_main: no such file or directory
Local symbolization failed for libpthread-2.15.so: stat /libpthread-2.15.so: no such file or directory
Some binary filenames not available. Symbolization may be incomplete.
Try setting PPROF_BINARY_PATH to the search path for local binaries.
Local symbolization failed for cppbench_server_main: stat cppbench_server_main: no such file or directory
Local symbolization failed for libpthread-2.15.so: stat /libpthread-2.15.so: no such file or directory
Some binary filenames not available. Symbolization may be incomplete.
Try setting PPROF_BINARY_PATH to the search path for local binaries.
Local symbolization failed for cppbench_server_main: stat cppbench_server_main: no such file or directory
Local symbolization failed for libpthread-2.15.so: stat /libpthread-2.15.so: no such file or directory
Some binary filenames not available. Symbolization may be incomplete.
Try setting PPROF_BINARY_PATH to the search path for local binaries.
````

I updated the comment to reflect that.